### PR TITLE
[ETCM-1045] Move getChainWeight to BlockchainReader

### PR DESCRIPTION
**Pull request by leo-bogastry**
# Description

In order to reduce the scope of the Blockchain class, getChainWeight was moved to BlockchainReader

# Testing
This branch was deployed in Staging

### CODE REVIEW

File: src/test/scala/io/iohk/ethereum/domain/BlockchainSpec.scala [hunk 1]
```diff
@@ -214,7 +214,7 @@ class BlockchainSpec extends AnyFlatSpec with Matchers with ScalaCheckPropertyCh
   }
 
   it should "return correct best block number after saving and rolling back blocks" in new TestSetup {
-    forAll(posIntGen(min = 1, max = maxNumberBlocksToImport)) { numberBlocksToImport =>
+    forAll(intGen(min = 1, max = maxNumberBlocksToImport)) { numberBlocksToImport =>
```
Comment (leo-bogastry) [inline src/test/scala/io/iohk/ethereum/domain/BlockchainSpec.scala]: I did not understand why this test was failing in a previous PR, the generated number sometimes where negative so I added the `posIntGen` that had a clause filtering negative numbers.
In this branch this test was failing again sometimes so, looking into it, I realized that now that the cache was removed in previous PR's the way the value `expectedPersistedBestBlock` (line 257) was being calculated made no sense and simplified the test

**Review by AurelienRichez — APPROVED** [review]

**Review by AnastasiiaL — APPROVED** [review]

File: src/test/scala/io/iohk/ethereum/ledger/LedgerTestSetup.scala [hunk 2]
```diff
@@ -479,7 +478,7 @@ trait EphemBlockchain extends TestSetupWithVmAndValidators with MockFactory {
 trait CheckpointHelpers {
   private val sampleCheckpoint = ObjectGenerators.fakeCheckpointGen(3, 3).sample.get
 
-  def getCheckpointBlock(parent: Block, difficulty: BigInt, checkpoint: Checkpoint = sampleCheckpoint): Block =
```
Comment (ghost) [inline src/test/scala/io/iohk/ethereum/ledger/LedgerTestSetup.scala:482]: @LeonorLunatech what is the reason for this change?

Comment (leo-bogastry) [reply src/test/scala/io/iohk/ethereum/ledger/LedgerTestSetup.scala:482]: The parameter `difficulty` was not being used

Comment (leo-bogastry) [reply src/test/scala/io/iohk/ethereum/ledger/LedgerTestSetup.scala:482]: We have several warnings being output when the code compile, and we are trying to take care of them. This was one of them
